pkg/cmd/gist/edit/edit.go

```diff
@@ -222,3 +259,74 @@ func updateGist(apiClient *api.Client, hostname string, gist *shared.Gist) error
 
 	return nil
 }
+
+func getFilesToAdd(file string, opts *EditOptions) (map[string]*shared.GistFile, error) {
+	cs := opts.IO.ColorScheme()
+
+	fileExists, err := fileExists(file)
+	if err != nil {
+		return nil, fmt.Errorf("%s %s", cs.Red("!"), err)
+	}
+
+	filesToAdd := map[string]*shared.GistFile{}
+
+	filename := filepath.Base(file)
+
+	if fileExists {
+		content, err := ioutil.ReadFile(file)
+		if err != nil {
+			return nil, fmt.Errorf("%s failed to read file %s: %w", cs.FailureIcon(), file, err)
+		}
+
+		if string(content) == "" {
+			return nil, fmt.Errorf("%s Contents can't be empty", cs.FailureIcon())
+		}
+
+		filesToAdd[filename] = &shared.GistFile{
+			Filename: filename,
+			Content:  string(content),
+		}
+		return filesToAdd, nil
+	} else {
```

Comment:
I've been playing around with this in my terminal and let's please go with my original request of not opening the editor to add a new file and instead just printing an error.

If a user mistypes a filename they are suddenly dropped into an editor which is very confusing and I don't think it's too much to ask that a user have a file already created that they want to add.

Reply:
oh you're right. I haven't tested that case. Makes sense. 👍🏼